File: solutions/ocaml/065-maximum-depth-of-binary-tree.ml
(*
  Problem 65: Maximum Depth of Binary Tree (LeetCode 104)
  Difficulty: Easy
  Language: OCaml
*)
let rec max_depth = function
  | Nil -> 0
  | Node (_, l, r) -> 1 + max (max_depth l) (max_depth r)

File: solutions/haskell/065-maximum-depth-of-binary-tree.hs
{-
  Problem 65: Maximum Depth of Binary Tree (LeetCode 104)
  Difficulty: Easy
  Language: Haskell
-}
maxDepth :: Tree a -> Int
maxDepth Nil = 0
maxDepth (Node _ l r) = 1 + max (maxDepth l) (maxDepth r)
